backend/de.metas.async/src/main/java/de/metas/async/api/impl/QueueDAO.java

```diff
@@ -204,12 +206,9 @@ public QueueDAO()
 		if (packageProcessorIds != null)
```

Comment:
shall we check for !packageProcessorIds.isEmpty() just to be on the safe side?

(even though that's already checked in that PackageQuery class)

Reply:
hmm, on the `PlainQueueDAO` we have a different behaviour, meaning  the filter returns `false` if no `packageProcessor` was provided (and logs a message, did the same for consistency => I feel like it might be a bad idea to let the query run without a `packageProcessorId` constraint if the array was empty)

Reply:
@cp-ps  Ok, but in that case, if `packageProcessorIds` is empty basically your query will always return no result.

In that case, I would suggest checking where WorkPackageQuery is instantiated and there make sure `packageProcessorIds` get a non-empty list.
You can even enforce this in WorkPackageQuery.setPackageProcessorIds setter.

In this method, in case the packageProcessorIds is empty I would return a "null" query because there is no point to execute a possible expensive query to the database when you know from the very beginning that no result will be returned.

wdyt?

Reply:
yep, right, changed accordingly 👍